genie-core/src/main/java/com/netflix/genie/core/services/impl/LocalJobRunner.java

```diff
@@ -203,6 +215,38 @@ public void submitJob(
         }
     }
 
+    private void createInitFailureDetailsFile(final String id, final Exception e) {
+        final long start = System.nanoTime();
+        try {
+            final File jobDir = new File(this.baseWorkingDirPath.getFile(), id);
+            if (jobDir.exists()) {
+                final File detailsFile = new File(jobDir, JobConstants.GENIE_INIT_FAILURE_MESSAGE_FILE_NAME);
+                final boolean detailsFileExists = !detailsFile.createNewFile();
+                if (detailsFileExists) {
+                    log.warn("Init failure details file exists");
+                }
+                try (final Writer writer = new OutputStreamWriter(new FileOutputStream(detailsFile), "UTF-8")) {
+
+                    writer.write("Job "  + id + " failed initialization due to: " + e.getMessage());
+                    writer.write(System.lineSeparator());
+                    writer.write("Exception: " + e.getClass().getCanonicalName());
+                    writer.write(System.lineSeparator());
+                    writer.write("Trace:");
+                    writer.write(System.lineSeparator());
+                    e.printStackTrace(new PrintWriter(writer));
+                    writer.write(System.lineSeparator());
+                }
+                log.info("Created init failure details file {}", detailsFile);
+            } else {
+                log.error("Could not create init failure details file, job directory does not exist");
+            }
+        } catch (Throwable t) {
+            log.error("Failed to create init failure details file", t);
```

Comment:
I assumed timer also includes a counter. And things like createJobDir also use a timer rather than a counter so just went for the same.

As for `Throwable` I really wanted to make sure that this function does not throw.